test/test_session.cpp

```diff
@@ -510,3 +510,70 @@ TORRENT_TEST(init_dht)
 		TEST_EQUAL(count, 1);
 	}
 }
+
+TORRENT_TEST(reopen_network_sockets)
+{
+	auto count_alerts = [](session& ses, int listen, int portmap)
+	{
+		int count_listen = 0;
+		int count_portmap = 0;
+		int num = 120; // this number is adjusted per version, an estimate
+		time_point const end_time = clock_type::now() + seconds(15);
+		while (true)
+		{
+			time_point const now = clock_type::now();
+			if (now > end_time)
+				break;
+
+			ses.wait_for_alert(end_time - now);
+			std::vector<alert*> alerts;
+			ses.pop_alerts(&alerts);
+			for (auto a : alerts)
+			{
+				std::printf("%d: [%s] %s\n", num, a->what(), a->message().c_str());
+				std::string const msg = a->message();
+				if (msg.find("successfully listening on") != std::string::npos)
+					count_listen++;
+				// upnp
+				if (msg.find("adding port map:") != std::string::npos)
```

Comment:
it's a bit dangerous to look for strings in the log :) I can't think of a better way though, short of implementing a dummy UPnP- and NATPMP router

Reply:
what?!! :D